service/controller/release/resource/argoapps/resource.go

```diff
@@ -0,0 +1,255 @@
+package argoapps
+
+import (
+	"context"
+	"fmt"
+
+	releasev1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	argoappv1alpha1 "github.com/giantswarm/argoapp/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/argoapp/pkg/argoapp"
+
+	"github.com/giantswarm/release-operator/v2/pkg/project"
+	"github.com/giantswarm/release-operator/v2/service/controller/key"
+)
+
+const (
+	Name = "argoapps"
+)
+
+var (
+	argoCDNamespace = "argocd"
+)
+
+type Config struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+}
+
+type Resource struct {
+	k8sClient k8sclient.Interface
+	logger    micrologger.Logger
+}
+
+func New(config Config) (*Resource, error) {
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	r := &Resource{
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return Name
+}
+
+func (r *Resource) ensureState(ctx context.Context) error {
+	var releases releasev1alpha1.ReleaseList
+	{
+		err := r.k8sClient.CtrlClient().List(
+			ctx,
+			&releases,
+		)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		releases = key.ExcludeDeletedRelease(releases)
+		releases = key.ExcludeUnusedDeprecatedReleases(releases)
+	}
+
+	var components map[string]releasev1alpha1.ReleaseSpecComponent
+	{
+		components = key.ExtractComponents(releases)
+	}
+
+	componentApps := map[string]argoappv1alpha1.Application{}
```

Comment:
Correct me if I'm wrong, but I don't see a reason for this not being a slice of `[]argoappv1alpha1.Application{}`

Reply:
Ack, remnant of previous implementation.